modules/checkout/tests/src/Functional/CheckoutOrderTest.php

```diff
@@ -0,0 +1,157 @@
+ *
+ * @preserveGlobalState disabled
+ */
+class CheckoutOrderTest extends BrowserTestBase {
+
+  use BlockCreationTrait;
+
+  /**
+   * The current user.
+   *
+   * @var \Drupal\Core\Session\AccountInterface;
+   */
+  protected $account;
+
+  /**
+   * The product.
+   *
+   * @var \Drupal\commerce_product\Entity\ProductInterface
+   */
+  protected $product;
+
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  public static $modules = ['system', 'field', 'user', 'text',
+    'entity', 'views', 'address', 'profile', 'commerce', 'inline_entity_form',
+    'commerce_price', 'commerce_store', 'commerce_product', 'commerce_cart',
+    'commerce_checkout', 'commerce_order',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $store = $this->createDefaultStore();
+
+    $variation = ProductVariation::create([
+      'type' => 'default',
+      'sku' => strtolower($this->randomMachineName()),
+      'price' => [
+        'amount' => 9.99,
+        'currency_code' => 'USD',
+      ],
+    ]);
+    $variation->save();
+
+    /** @var \Drupal\commerce_product\Entity\ProductInterface $product */
+    $this->product = Product::create([
+      'type' => 'default',
+      'title' => 'My product',
+      'variations' => [$variation],
+      'stores' => [$store],
+    ]);
+    $this->product->save();
+
+    user_role_grant_permissions(RoleInterface::ANONYMOUS_ID, [
+      'access content',
+      'access checkout',
+    ]);
+  }
+
+  /**
+   * Tests than an order can go through checkout steps.
+   */
+  public function testGuestOrderCheckout() {
+    $this->drupalGet($this->product->toUrl()->toString());
+    $this->submitForm([], 'Add to cart');
+    $cart_link = $this->getSession()->getPage()->findLink('your cart');
+    $cart_link->click();
+    $this->submitForm([], 'Checkout');
+    $this->submitForm([], 'Continue as Guest');
+    $this->submitForm([
+      'contact_information[email]' => 'guest@example.com',
+      'contact_information[email_confirm]' => 'guest@example.com',
+      'billing_information[address][0][recipient]' => $this->randomString(),
+      'billing_information[address][0][organization]' => $this->randomString(),
+      'billing_information[address][0][address_line1]' => $this->randomString(),
+      'billing_information[address][0][locality]' => $this->randomString(),
+    ], 'Continue to review');
+    $this->assertSession()->pageTextContains('Contact information');
+    $this->assertSession()->pageTextContains('Billing information');
+    $this->submitForm([], 'Pay and complete purchase');
+    $this->assertSession()->pageTextContains('Your order number is 1. You can view your order on your account page when logged in.');
+
+    $this->saveHtmlOutput();
+  }
+
+  /**
+   * Creates and sets the default store.
+   *
+   * @return \Drupal\commerce_store\Entity\StoreInterface
+   *    The store.
+   */
+  protected function createDefaultStore() {
```

Comment:
I'd like to get this committed, then move to an attribute in `Drupal\Tests\commerce_store\CreateStoreAttribute` or something